src/models/HTTPResponse.ts

```diff
@@ -0,0 +1,25 @@
+import {APIGatewayProxyResult} from "aws-lambda";
+export class HTTPResponse extends Error implements APIGatewayProxyResult {
+  public body: any;
+  public statusCode: number;
+  public readonly headers: any;
+
+  /**
+   * Constructor for the HTTPResponse class
+   * @param statusCode the HTTP status code
+   * @param body - the response body
+   * @param headers - optional - the response headers
+   */
+  constructor(statusCode: number, body: any, headers = {}) {
+    super();
+
+    if (headers) { this.headers = headers; }
```

Comment:
I know this is everywhere, but just realised - does this need the "if" if headers defaults to an empty object?

Reply:
good point ... slipped my eyes ... `if {header)` will always evaluate to truthy ... due to default ... hence the `if` check is unnecessary ... the reason for default empty object is understandable